gradoop-flink/src/main/java/org/gradoop/flink/algorithms/gelly/randomjump/functions/VisitedGellyEdgesWithLongIdToGradoopIdJoin.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright © 2014 - 2018 Leipzig University (Database Research Group)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradoop.flink.algorithms.gelly.randomjump.functions;
+
+import org.apache.flink.api.common.functions.JoinFunction;
+import org.apache.flink.api.java.functions.FunctionAnnotation;
+import org.apache.flink.api.java.tuple.Tuple2;
+import org.gradoop.common.model.impl.id.GradoopId;
+
+/**
+ * Joins an edge index with its index-to-GradoopId-mapping to replace the index with the GradoopId.
+ */
+@FunctionAnnotation.ReadFieldsSecond("f1")
```

Comment:
The field will be forwarded, so use that annotation.

Reply:
Done

Reply:
If you don't specify the target, the same field is assumed for source and target.
This will fail here since the target is a `GradoopId` which does not have  a `f1` field.
Try
```java
@FunctionAnnotation.ForwardedFieldsFirst("f1->*")
```

Reply:
Aaah, I see. My fault, got it now. Has to be `@FunctionAnnotation.ForwardedFieldsSecond("f1->*")`,  though ;)